File: packages/typescript/src/surfaces/conversation.ts
import type { YaiEnvelope } from '../envelope';
import { YAI_OPERATIONS } from '../operations';
import { TransportBackedSurface } from './base';

export class ConversationSurface extends TransportBackedSurface {
  current(request?: unknown): Promise<YaiEnvelope> {
    return this.invoke(YAI_OPERATIONS.conversationCurrent, request);
  }

  messagesSend(request?: unknown): Promise<YaiEnvelope> {
    return this.invoke(YAI_OPERATIONS.conversationMessagesSend, request);
  }
}

File: packages/rust/src/surfaces/conversation.rs
use serde_json::Value;

use crate::envelope::YaiEnvelope;
use crate::operations::YAI_OPERATIONS;
use crate::status::YaiResult;
use crate::transport::YaiTransport;

pub struct ConversationSurface<TTransport: YaiTransport> {
    transport: TTransport,
}

impl<TTransport: YaiTransport> ConversationSurface<TTransport> {
    pub fn new(transport: TTransport) -> Self {
        Self { transport }
    }

    pub fn current(&self) -> YaiResult<YaiEnvelope<Value>> {
        self.transport
            .invoke::<Value>(YAI_OPERATIONS.conversation_current, None)
    }

    pub fn messages_send(&self, request: Option<Value>) -> YaiResult<YaiEnvelope<Value>> {
        self.transport
            .invoke::<Value>(YAI_OPERATIONS.conversation_messages_send, request)
    }
}
